src/nrniv/partrans.cpp

```diff
@@ -285,7 +285,11 @@ void nrnmpi_source_var() {
 	alloclists();
 	is_setup_ = false;
 	double* psv = hoc_pgetarg(1); // but might not be a voltage
-	sgid_t sgid = (sgid_t)(*getarg(2));
+	double x = *getarg(2);
+	if (x < 0) {
+		hoc_execerr_ext("source_var sgid must be >= 0: arg 2 is %g\n", x);
+	}
+	sgid_t sgid = (sgid_t)x;
```

Comment:
nitpick : I assume x won't be  0 < x < 1 in which case this will still fail as sgid will be 0?

Reply:
sgid = 0 is acceptable.